src/services/fcmToken/sendChatNotification.js

```diff
@@ -0,0 +1,43 @@
+const {messaging} = require('firebase-admin');
+const UsersFunction = require('../../databases/functions/users');
+const FcmTokenFunction = require('../../databases/functions/fcmToken');
+const {User, FcmToken} = require('../../databases/models');
+
+const sendChatNotification = async (userTargetId, payload) => {
+  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId, {
+    shouldReturnObject: true
+  });
+
+  const userTargetToken = await FcmTokenFunction.findAllTokenByUserId(
+    FcmToken,
+    signedUser?.user_id
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is using optional chaining (`?.`) which is a great way to avoid `undefined` errors. However, it's important to handle the case where `signedUser` or `signedUser.user_id` might be undefined. If `signedUser` is undefined, then `signedUser?.user_id` will also be undefined and this could lead to unexpected behavior in `FcmTokenFunction.findAllTokenByUserId`. Consider adding error handling for this scenario.

```diff
+  if (!signedUser || !signedUser.user_id) {
+    throw new Error('User not found');
+  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->